state_test.go

```diff
@@ -0,0 +1,171 @@
+package mls
+
+import (
+	"fmt"
+	"testing"
+)
+
+var (
+	groupId   = []byte{0x01, 0x02, 0x03, 0x04}
+	userId    = []byte{0x04, 0x05, 0x06, 0x07}
+	suite     = P256_SHA256_AES128GCM
+	scheme    = Ed25519
+	groupSize = 5
+	setupDone = false
+
+	identityPrivs  []SignaturePrivateKey
+	credentials    []Credential
+	initPrivs      []HPKEPrivateKey
+	clientInitKeys = make([]ClientInitKey, groupSize)
+	states         []State
+
+	testMessage = unhex("1112131415")
+)
+
+func setup(t *testing.T) {
+	if setupDone {
+		return
+	}
+
+	for i := 0; i < groupSize; i++ {
+		// cred gen
+		sigPriv, _ := scheme.Generate()
+		basicCredential = &BasicCredential{
+			Identity:           userId,
+			SignatureScheme:    scheme,
+			SignaturePublicKey: sigPriv.PublicKey,
+		}
+		credentialBasic = Credential{Basic: basicCredential, privateKey: &sigPriv}
+
+		//cik gen
+		cik, err := newClientInitKey(suite, &credentialBasic)
+		assertNotError(t, err, "newClientInitKey error")
+		// save all the materials
+		identityPrivs = append(identityPrivs, sigPriv)
+		credentials = append(credentials, credentialBasic)
+		initPrivs = append(initPrivs, ikPriv)
+		clientInitKeys[i] = *cik
+		cik = nil
+		//dump(clientInitKeys)
+	}
+	setupDone = true
+}
+
+func dump(ciks []ClientInitKey) {
+	fmt.Println("---- DUMP -----")
+	for _, cik := range ciks {
+		fmt.Printf("priv %x pub %x\n", cik.privateKey.Data, cik.InitKey.Data)
+	}
+}
+
+func TestState_TwoPerson(t *testing.T) {
+	setup(t)
+	// creator's state
+	// dump(clientInitKeys)
+	first0 := newEmptyState(groupId, suite, initPrivs[0], credentials[0])
+	// add the second participant
+	add := first0.add(clientInitKeys[1])
```

Comment:
Should this use `negotiate()`?  Maybe we can get rid of `newEmptyState`.

Reply:
will tackle this in the test update PR